File: src/main/java/pl/yalgrin/playnite/simplesync/web/CategoryResource.java
package pl.yalgrin.playnite.simplesync.web;

import lombok.RequiredArgsConstructor;
import org.springframework.web.bind.annotation.*;
import pl.yalgrin.playnite.simplesync.dto.CategoryDTO;
import pl.yalgrin.playnite.simplesync.helper.SingleExecutorHelper;
import pl.yalgrin.playnite.simplesync.service.CategoryService;
import reactor.core.publisher.Mono;

@RestController
@RequestMapping("/api/category")
@RequiredArgsConstructor
public class CategoryResource {
    private final CategoryService service;
    private final SingleExecutorHelper singleExecutorHelper;

    @GetMapping("/{id}")
    public Mono<CategoryDTO> getCategory(@PathVariable Long id) {
        return service.findById(id);
    }

    @PostMapping("/save")
    public Mono<CategoryDTO> saveCategory(@RequestBody CategoryDTO dto, @RequestParam String clientId) {
        return singleExecutorHelper.runOnExecutor(service.saveObject(dto, clientId));
    }

    @PostMapping("/delete")
    public Mono<Void> deleteCategory(@RequestBody CategoryDTO dto, @RequestParam String clientId) {
        return singleExecutorHelper.runOnExecutor(service.deleteObject(dto, clientId));
    }
}

File: src/main/kotlin/pl/yalgrin/playnite/simplesync/web/library/CategoryResource.kt
package pl.yalgrin.playnite.simplesync.web.library

import org.springframework.web.bind.annotation.*
import pl.yalgrin.playnite.simplesync.helper.SingleExecutorHelper
import pl.yalgrin.playnite.simplesync.library.dto.CategoryDTO
import pl.yalgrin.playnite.simplesync.service.objects.CategoryService
import reactor.core.publisher.Mono

@RestController
@RequestMapping("/api/category")
class CategoryResource(
    private val service: CategoryService,
    private val singleExecutorHelper: SingleExecutorHelper
) {

    @GetMapping("/{id}")
    fun getCategory(@PathVariable id: Long): Mono<CategoryDTO> {
        return service.findById(id)
    }

    @PostMapping("/save")
    fun saveCategory(@RequestBody dto: CategoryDTO): Mono<CategoryDTO> {
        return singleExecutorHelper.runOnExecutor(service.saveObject(dto))
    }

    @PostMapping("/delete")
    fun deleteCategory(@RequestBody dto: CategoryDTO): Mono<Void> {
        return singleExecutorHelper.runOnExecutor(service.deleteObject(dto))
    }
}